openmls/src/group/mls_group/mod.rs

```diff
@@ -72,50 +72,107 @@ implement_persistence!(
     mls_version
 );
 
-/// Public `MlsGroup` functions.
-impl MlsGroup {
-    pub fn new(
-        id: &[u8],
-        ciphersuite_name: CiphersuiteName,
-        backend: &impl OpenMlsCryptoProvider,
-        key_package_bundle: KeyPackageBundle,
-        config: MlsGroupConfig,
-        psk_option: impl Into<Option<PskSecret>>,
-        version: impl Into<Option<ProtocolVersion>>,
-    ) -> Result<Self, MlsGroupError> {
-        debug!("Created group {:x?}", id);
-        trace!(" >>> with {:?}, {:?}", ciphersuite_name, config);
-        let group_id = GroupId { value: id.into() };
-        let ciphersuite = Config::ciphersuite(ciphersuite_name)?;
-        let tree = RatchetTree::new(backend, key_package_bundle);
-        // TODO #483: Implement extensions
-        let extensions: Vec<Extension> = Vec::new();
+/// Builder for [`MlsGroup`].
+pub struct MlsGroupBuilder {
+    key_package_bundle: KeyPackageBundle,
+    group_id: Option<GroupId>,
+    config: Option<MlsGroupConfig>,
+    psk: Option<PskSecret>,
+    version: Option<ProtocolVersion>,
+    required_capabilities: Option<RequiredCapabilitiesExtension>,
+}
+
+impl MlsGroupBuilder {
+    /// Create a new [`MlsGroupBuilder`].
+    pub fn new(key_package_bundle: KeyPackageBundle) -> Self {
```

Comment:
Is there a specific reason why the two mandatory parameters `KeyPackageBundle` and `OpenMlsCryptoProvider` are spread over two different functions? Previously we used mandatory `with_*` functions (with corresponding `struct`s) to make it more explicit. I'm not very hung up on what we end up using, but I think we should harmonise this across the API.

Reply:
The `OpenMlsCryptoProvider` is an `impl` argument that's impossible to store. That's why it has to be passed to every function that needs it. Here only the `build` needs it.